src/test/java/user/sort/SortAlgorithmsTest.java

```diff
@@ -0,0 +1,110 @@
+        SortResponseModel responseModel = sortAlgorithm.sort(reviews);
+        assertEquals(correctlySorted, responseModel.getSortedOutput());
+    }
+
+    @Test
+    void testNewestSort(){
+        NewestSort sortAlgorithm = new NewestSort();
+        SortResponseModel responseModel = sortAlgorithm.sort(reviews);
+        assertEquals(correctlySorted, responseModel.getSortedOutput());
+    }
+
+    @Test
+    void testSortTwoValuesEqualCase(){
+        review5 = new Review("3", "user3", "", 3, "company1", new Date(2022, Calendar.JUNE, 24), 3, 3, new ArrayList<>());
+        reviews.add(review5);
+        ArrayList<Review> reviews1 = new ArrayList<Review>(reviews);
+        ArrayList<Review> reviews2 = new ArrayList<Review>(reviews);
+        ArrayList<Review> reviews3 = new ArrayList<Review>(reviews);
+
+        ArrayList<Integer> correctlySortedHelpful = new ArrayList<Integer>();
+        correctlySortedHelpful.add(115);
+        correctlySortedHelpful.add(5);
+        correctlySortedHelpful.add(3);
+        correctlySortedHelpful.add(3);
+        correctlySortedHelpful.add(0);
+        ArrayList<Integer> correctlySortedHighestRating = new ArrayList<Integer>();
+        correctlySortedHighestRating.add(10);
+        correctlySortedHighestRating.add(5);
+        correctlySortedHighestRating.add(3);
+        correctlySortedHighestRating.add(3);
+        correctlySortedHighestRating.add(0);
+        ArrayList<Date> correctlySortedNewest = new ArrayList<Date>();
+        correctlySortedNewest.add(new Date(2022, Calendar.SEPTEMBER, 24));
+        correctlySortedNewest.add(new Date(2022, Calendar.JULY, 24));
+        correctlySortedNewest.add(new Date(2022, Calendar.JUNE, 24));
+        correctlySortedNewest.add(new Date(2022, Calendar.JUNE, 24));
+        correctlySortedNewest.add(new Date(2022, Calendar.MARCH, 24));
+
+        SortResponseModel helpfulResponse = new HelpfulSort().sort(reviews1);
+        SortResponseModel highestRatingResponse = new HighestRatingSort().sort(reviews2);
+        SortResponseModel newestResponse = new NewestSort().sort(reviews3);
+
+        ArrayList<Integer> sortedValuesHelpful = new ArrayList<Integer>();
+        ArrayList<Integer> sortedValuesHighestRating = new ArrayList<Integer>();
+        ArrayList<Date> sortedValuesNewest = new ArrayList<Date>();
+
+        for (Review review : helpfulResponse.getSortedOutput()){
+            sortedValuesHelpful.add(review.getNumLikes());
+        }
+        for (Review review : highestRatingResponse.getSortedOutput()){
+            sortedValuesHighestRating.add(review.getRating());
+        }
+        for (Review review : newestResponse.getSortedOutput()){
+            sortedValuesNewest.add(review.getDatePosted());
+        }
+
+        assertEquals(correctlySortedHelpful, sortedValuesHelpful);
+        assertEquals(correctlySortedHighestRating, sortedValuesHighestRating);
+        assertEquals(correctlySortedNewest, sortedValuesNewest);
+    }
```

Comment:
Would it be better to directly compare two ArrayLists of review objects, instead of the values?

Reply:
No, in this case I have to be comparing ArrayLists of the values. I'm trying to test for two Reviews with the same values being sorted to be next to each other, so I have to show that it can work regardless of which Review is put first. However, the sorting algorithms will sort the same Review to be placed first every time, so comparing the Reviews I can't show that the placement doesn't matter, since trying to compare with the values swapped would result in a failure.